examples/web/index.html

```diff
@@ -501,6 +501,339 @@ <h2 class="typography-heading-medium-text">Alert</h2>
+                );
+              }
+
+              return null;
+            };
+
+            const closeOpenHeaderPanes = () => {
+              getOpenHeaderPanes().forEach((headerPaneEl) => toggleHeaderPane(headerPaneEl));
+            };
+
+            const toggleHeaderPane = (targetEl) => {
+              if (!targetEl) {
+                console.warn('👻 Boo…! Target header pane does not exist. Maybe you forgot to prefix the "data-target" selector with "#"? ');
+                return;
+              }
+
+              const headerEl = targetEl.parentElement;
+              const allHeaderToggleEls = headerEl.querySelectorAll(HEADER_TOGGLE_SELECTOR);
+              const openHeaderPaneEls = getOpenHeaderPanes();
+
+              if (targetEl.classList.contains(OPEN_CLASSNAME)) {
+                targetEl.classList.remove(OPEN_CLASSNAME);
+                allHeaderToggleEls.forEach((toggleEl) => {
+                  if (headerEl.querySelector(toggleEl.dataset.target) === targetEl) {
+                    toggleEl.setAttribute('aria-expanded', 'false');
+                  }
+                });
+
+                if (openHeaderPaneEls.length === 1) {
+                  document.removeEventListener('keydown', handleKeyPress);
+                  window.removeEventListener('resize', handleWindowResize);
+
+                  // TODO #DS-190
+                  // window.removeEventListener('click', handleClickOutside);
+                }
+              } else {
+                targetEl.classList.add(OPEN_CLASSNAME);
+                allHeaderToggleEls.forEach((toggleEl) => {
+                  if (headerEl.querySelector(toggleEl.dataset.target) === targetEl) {
+                    toggleEl.setAttribute('aria-expanded', 'true');
+                  }
+                });
+
+                if (!openHeaderPaneEls.length) {
+                  document.addEventListener('keydown', handleKeyPress);
+                  window.addEventListener('resize', handleWindowResize);
+
+                  // TODO #DS-190
+                  // window.addEventListener('click', handleClickOutside);
+                }
+              }
+            };
+
+            const handleHeaderToggleClick = (event) => {
+              toggleHeaderPane(document.querySelector(event.currentTarget.dataset.target));
+            };
+
+            // TODO #DS-190
+            // const handleClickOutside = (event) => {};
+
+            const handleKeyPress = (event) => {
+              // Do nothing if the event was already processed.
+              if (event.defaultPrevented) {
+                return;
+              }
+
+              if (event.key === 'Esc' || event.key === 'Escape') {
+                closeOpenHeaderPanes();
+              }
+
+              // Cancel the default action to avoid it being handled twice.
+              event.preventDefault();
+            };
+
+            const handleWindowResize = (event) => {
```

Comment:
This closes the desktop dialog when resizing. Is this on purpose?

Reply:
Since I don't distinguish between the responsive actions bar and the dialog in this script, I don't know how to avoid that. And we need to "close" the actions bar on desktop so it doesn't open again when you e.g. rotate your device.

I think the only way is to split resize handling of actions bar and header dialog (which could be done through a new data attribute, or a new value of `data-toggle`: `header-actions` or `header-dialog` instead of current `header` for both).

Reply:
This may be better covered in #250 which officially introduces JS handlers for our components.